crates/viewer/re_space_view_spatial/src/mesh_cache.rs

```diff
@@ -36,7 +36,13 @@ pub struct MeshCache(HashMap<RowId, HashMap<MeshCacheKey, Option<Arc<LoadedMesh>
 /// Either a [`re_types::archetypes::Asset3D`] or [`re_types::archetypes::Mesh3D`] to be cached.
 #[derive(Debug, Clone, Copy)]
 pub enum AnyMesh<'a> {
-    Asset(&'a re_types::archetypes::Asset3D),
+    Asset {
+        asset: &'a re_types::archetypes::Asset3D,
+
+        /// If there are any textures associated with that asset (albedo etc), they use this
+        /// hash for texture manager lookup.
+        texture_key: u64,
```

Comment:
See `assets3d.rs` visualizer, this isn't needed.
The only reason the `Mesh` variant needs this is to account for the fact that there's additional archetypes on the same entity that are taken into consideration